### PR TITLE
declcfg/load: improvements

### PR DESCRIPTION
**Pull request by stevekuznetsov**

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Sits on top of #1094 
Fixes #1103 
Fixes #1102

**Comment by stevekuznetsov** [comment]
@joelanford interestingly even with 16 logical cores I only see an 8x improvement with real filesystem access:

```
$ go test -bench=BenchmarkLoadFS ./alpha/declcfg/... -benchtime=10s
goos: linux
goarch: amd64
pkg: github.com/operator-framework/operator-registry/alpha/declcfg
cpu: 11th Gen Intel(R) Core(TM) i7-11700K @ 3.60GHz
BenchmarkLoadFS/1_routines-16         	      15	 746241287 ns/op
BenchmarkLoadFS/16_routines-16        	     100	 102231403 ns/op
BenchmarkLoadFS/32_routines-16        	     100	 103457630 ns/op
PASS
ok  	github.com/operator-framework/operator-registry/alpha/declcfg	79.606s
```

**Comment by stevekuznetsov** [comment]
> 1.  Looks like there's a unit test that needs the context passed in.

Oops, broke that in the rebase. Fixed.

> 2. Are there any other functions in the declcfg package that would make sense to also go ahead and add a context parameter to?

I don't know off the top of my head :)

File: alpha/template/composite/builder.go [hunk 1]
```diff
@@ -311,14 +311,14 @@ func validate(builderCfg BuilderConfig, dir string) error {
...
-	if err := config.Validate(os.DirFS(path)); err != nil {
+	if err := config.Validate(ctx, os.DirFS(path)); err != nil {
 		return fmt.Errorf("validation failure in path %q: %v", path, err)
 	}
 	return nil
 }
 
 func build(dcfg *declcfg.DeclarativeConfig, outPath string, outType string) error {
-	// create the destination for output, if it does not exist
+	// create the destination for output, if it does not existalpha/declcfg/load_benchmark_test.go
```
Comment (grokspawn) [inline alpha/template/composite/builder.go:321]: ???

Comment (stevekuznetsov) [reply alpha/template/composite/builder.go]: Copy pasta fail - nice catch

**Review by grokspawn — APPROVED** [review]

**Pull request merged**